src/services/codefixes/importFixes.ts

```diff
@@ -356,6 +358,35 @@ function createImportAdderWorker(sourceFile: SourceFile | FutureSourceFile, prog
         }
     }
 
+    function addImportForExternalModuleSymbol(symbol: Symbol, symbolName: string, isValidTypeOnlyUseSite: boolean, referenceImport?: ImportOrRequireAliasDeclaration) {
+        const useRequire = shouldUseRequire(sourceFile, program);
+        const moduleSpecifier = moduleSpecifiers.getLocalModuleSpecifierBetweenFileNames(
+            sourceFile,
+            Debug.checkDefined(symbol.valueDeclaration).getSourceFile().fileName,
+            compilerOptions,
+            createModuleSpecifierResolutionHost(program, host),
+        );
+        let info: FixInfo = {
+            fix: { kind: ImportFixKind.AddNew, importKind: ImportKind.Namespace, addAsTypeOnly: isValidTypeOnlyUseSite ? AddAsTypeOnly.Allowed : AddAsTypeOnly.NotAllowed, useRequire, moduleSpecifierKind: undefined, moduleSpecifier },
+            symbolName,
+            errorIdentifierText: undefined,
+        };
+        if (referenceImport && isValidTypeOnlyUseSite && isImportClause(referenceImport)) {
+            info = {
+                ...info,
+                fix: {
+                    kind: ImportFixKind.AddNew,
+                    importKind: isNamespaceImport(referenceImport) ? ImportKind.Namespace : ImportKind.Default,

```

Comment:
This condition an never be true based on the if statement it's in. Also, it looks like we don't use the `referenceImport` to decide between default/namespace unless `isValidTypeOnlyUseSite` is true, which is an unrelated piece of information. I think all this code can be reduced to a single assignment to `info` with conditional properties, which would help untangle the two separate things you're trying to control.